### PR TITLE
Bloch cleanup dev

### PR DESCRIPTION
**Pull request by AGaliciaMartinez**
**Description**
Clean-up of the point method in the `Bloch` class. This PR does not add any new feature but it does:
- Improve readability and logic in the point methods for the Bloch class. 
- Extended test coverage of the point methods.
- Now add_points raises error if the arguments `points` or `meth` are wrong.

**Changelog**
Clean-up of the point methods in the `Bloch` class.

### CODE REVIEW

File: qutip/bloch.py [hunk 1]
```diff
@@ -798,53 +784,54 @@ def plot_vectors(self):
...
-                dist, indperm = zip(*zipped)
-                indperm = array(indperm)
+        for k, points in enumerate(self.points):
+            points = np.asarray(points)
+            num_points = points.shape[1]
+
+            dist = np.linalg.norm(points, axis=0)
+            if not np.allclose(dist, dist[0], rtol=1e-12):
+                indperm = np.argsort(dist)
+                points = points[:, indperm]
```
Comment (AGaliciaMartinez) [inline qutip/bloch.py:796]: This part of the code was also there before. I seems to sort points based on their distance to the bloch centre, but I am not sure why you would want to do so.... It is neither documented nor tested.

**Comment by AGaliciaMartinez** [comment]
I was surprised to see that the color argument was added long time ago for the point methods. These were not ported to master. These changes break backwards compatibility as code that used `point_colors` will not work anymore. I wonder if this change of variable name is something we want to keep as it should be possible to keep the behavior backwards compatible AND include the new colors argument (which I think is a very necessary feature that motivated me to clean the code in master in the first place, I later realized that this was already added in dev.major).

**Comment by Ericgig** [comment]
If my memory is good, `color` was added in a PR for GSoC application. It fixed the related issue, but still needed some polishing...  If you can fix any issue with, it would be appreciated.

File: qutip/bloch.py (hunk 1 above)
Comment (hodgestar) [reply qutip/bloch.py:795]: I suspect this is related to the behaviour of older versions of matplotlib. Before matplotlib 3.5 the rendering order of Axes3D was determined by the zordering of the Artist objects, which is a concept that only made sense for 2D rendering. In 3.5, Axes3D gained a "computed_zorder" option that is now the default and which should order everything correctly. See https://matplotlib.org/stable/api/_as_gen/mpl_toolkits.mplot3d.axes3d.Axes3D.html.

I would vote that we remove this reordering and live with possibly suboptimal rendering in old versions of matplotlib.

File: qutip/bloch.py [hunk 2]
```diff
@@ -799,52 +785,52 @@ def plot_vectors(self):
...
-                                  color=self.point_color[k][0], marker=marker)
+                indperm = np.arange(num_points)
+
+            s = self.point_size[mod(k, len(self.point_size))]
+            marker = self.point_marker[mod(k, len(self.point_marker))]
+            user_color = self.point_color[k]
+            style = self.point_style[k]
+            if user_color is not None:
+                color = user_color
+            elif user_color is None and self.point_style[k] in ['s', 'l']:
```
Comment (hodgestar) [inline qutip/bloch.py:805]:
```suggestion
            elif self.point_style[k] in ['s', 'l']:
```
`user_color` is always `None` in the `elif`.

**Review by hodgestar — CHANGES_REQUESTED** [review]
Thanks @AGaliciaMartinez! I left a couple of small suggestions but it mostly looks good to me. I suggest that we leave sorting out the point_color mismatch with master and the removal of the distance ordering for separate PRs (assuming we decide to fix those things).

**Comment by AGaliciaMartinez** [comment]
I agree on leaving this PR as it is. @hodgestar, I committed your suggestions and removed the `user_color` temporary variable (was not really necessary).

**Review by hodgestar — APPROVED** [review]
Approved. Happy for it to be merged once tests pass.

**Pull request merged**